packages/auth/src/accessToken/service.ts

```diff
@@ -78,10 +92,44 @@ async function revoke(
 ): Promise<Error | undefined> {
   const token = await AccessToken.query(deps.knex).findById(id)
   if (token) {
-    if (!isTokenExpired(token)) {
-      await token.$query(deps.knex).patch({ expiresIn: 1 })
+    if (!token.revoked && !isTokenExpired(token)) {
+      await token.$query(deps.knex).patch({ revoked: true })
```

Comment:
Is there a reason we are keeping the old tokens and are setting the `revoked` property? Why not cleaning them up, together with the expired tokens?

Reply:
According to the spec:
`An AS MUST NOT honor a rotation request for an access token that has been revoked or otherwise disabled`

So I think we have to use the `revoked` property in order to distinguish between tokens that have expired and tokens that were explicitly revoked

Reply:
But if the token doesn't exist anymore because it expired or was revoked, it can't honor the rotation request because it can't find the original token. Maybe I'm misunderstanding, though. 🤔

Reply:
Sorry, I should've also included this from the spec:

`Note that in many cases, the access token will have expired for
   regular use.  To facilitate token rotation, the AS SHOULD honor the
   rotation request of the expired access token since it is likely that
   the client instance is attempting to refresh the expired token.`

So it seems like we have to keep the token around to be able to determine whether it has been revoked or just expired

Reply:
Makes sense. We have an issue for that now. #385

Reply:
@sabineschaller After rereading the spec it seems like we do need to have a `revoked` field rather than just deleting or setting tokens to be expired:

`If the access token has already expired or has been revoked through other means, the AS SHOULD honor the revocation request to the token management URI as valid, since the end result is still the token not being usable.`

If we delete the token (before whatever time we use in #385 at least) then we can't honor the _revocation_ request.
But if we keep the token and just make it expired, then we can't know whether or not to honor the _rotation_ request.

Let me know what you think

Reply:
What if on revocation, the API didn't return a 404 for tokens that weren't found but always a 204 (given that auth is correct, i.e. no 401)? That way, the end result is still the same for the user. Token found - delete token - 204. Token not found - do nothing - 204. The spec doesn't mention a 404. We may just have put it in there without thinking about it.